comp/forwarder/defaultforwarder/forwarder.go

```diff
@@ -31,19 +33,46 @@ type provides struct {
 }
 
 func newForwarder(dep dependencies) provides {
-	return NewForwarder(dep.Config, dep.Log, dep.Lc, true, dep.Params)
+	options := creationOptions(dep.Params, dep.Config, dep.Log)
+	return NewForwarder(dep.Config, dep.Log, dep.Lc, true, options, dep.Params.useNoopForwarder)
+}
+
+func creationOptions(params Params, config config.Component, log log.Component) *Options {
+	var options *Options
+	if !params.withResolver {
+		options = NewOptions(config, log, getMultipleEndpoints(config, log))
+	} else {
+		keysPerDomain := getMultipleEndpoints(config, log)
+		options = NewOptionsWithResolvers(config, log, resolver.NewSingleDomainResolvers(keysPerDomain))
+	}
+	if disableAPIKeyChecking, ok := params.disableAPIKeyCheckingOverride.Get(); ok {
```

Comment:
What if `!ok`? Is it OK do not do anything?

Reply:
We want to override `options.DisableAPIKeyChecking` only if `WithFeatures` was called. Comment added.